frontend/src/components/pages/VolunteerDashboard/ShiftCard.tsx

```diff
@@ -0,0 +1,129 @@
+import { Box, Button, Stack, Text } from "@chakra-ui/react";
+import { format, parse } from "date-fns";
+import React, { useEffect, useState } from "react";
+import { useHistory } from "react-router-dom";
+
+import DonorAPIClient from "../../../APIClients/DonorAPIClient";
+import * as Routes from "../../../constants/Routes";
+import {
+  CheckInWithShiftType,
+  ScheduleWithShiftType,
+  ShiftType,
+} from "../../../types/VolunteerTypes";
+import CardField from "../../common/CardField";
+import { getShiftColor } from "./types";
+
+interface CheckInOrScheduleProps {
+  donorId?: string;
+  isPickup?: boolean;
+  pickupLocation?: string;
+  volunteerTime?: string;
+  startTime?: string;
+  startDate?: string;
+  notes: string;
+  type: ShiftType;
+}
```

Comment:
wouldn't the `id` need to be exposed here as well if we want to eventually be able to view details of this card?